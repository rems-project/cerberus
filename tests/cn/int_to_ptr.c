void* cast(unsigned long long addr)
/*@
ensures
    addr == 0u64 && is_null(return) || addr != 0u64 && has_alloc_id(return) && (u64) return == addr;
@*/
{
    return (void*)addr;
}

int main()
{
    int x = 0;
    void* p = cast((unsigned long long)&x);
    /*@ assert ((u64) &x == 0u64 || has_alloc_id(p) && (u64) p == (u64) &x); @*/
}
