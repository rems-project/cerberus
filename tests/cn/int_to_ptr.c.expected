return code: 0
[1/2]: cast -- pass
[2/2]: main -- pass
